### PR TITLE
Exposes alarm/health information in "etcdctl endpoint status"

### PR DESCRIPTION
**Pull request by dvonthenen**
Implements alarm/health information in "etcdctl endpoint status"
Addresses issue https://github.com/coreos/etcd/issues/9163

Tested on a local cluster. Provided mock errors to ensure formatting is correct.

### CODE REVIEW

File: etcdserver/api/v3rpc/maintenance.go [hunk 1]
```diff
@@ -38,6 +39,7 @@ type BackendGetter interface {
 }
 
 type Alarmer interface {
+	Alarms() []*pb.AlarmMember
```
Comment (xiang90) [inline etcdserver/api/v3rpc/maintenance.go]: can we document this public interface? I know there are existing undocumented methods, but we do need to improve the codebase in general :)

File: Documentation/dev-guide/apispec/swagger/rpc.swagger.json [hunk 2]
```diff
@@ -2177,7 +2184,7 @@
           "format": "uint64"
         },
         "raftAppliedIndex": {
-          "description": "appliedIndex is the current raft applied index of the responding member.",
+          "description": "raftAppliedIndex is the current raft applied index of the responding member.",
```
Comment (xiang90) [inline Documentation/dev-guide/apispec/swagger/rpc.swagger.json]: can we fix this in a separate pr? thanks. i want to keep the commit as clean as possible.

Comment (dvonthenen) [reply Documentation/dev-guide/apispec/swagger/rpc.swagger.json]: I didn't notice this change. Will have a subsequent PR for this. Not sure why this didn't get updated in the last PR when I did the proto gen.

File: etcdserver/api/v3rpc/maintenance.go (hunk 1 above)
Comment (dvonthenen) [reply etcdserver/api/v3rpc/maintenance.go]: Yep. This is an existing interface that I just added to this Alarmer interface. Will document it.

File: etcdserver/api/v3rpc/maintenance.go [hunk 3]
```diff
@@ -38,6 +39,9 @@ type BackendGetter interface {
 }
 
 type Alarmer interface {
+	//Alarms is implemented in Server interface located in etcdserver/server.go
```
Comment (xiang90) [inline etcdserver/api/v3rpc/maintenance.go]: put a space between `//` and `Alarms`

File: etcdserver/etcdserverpb/rpc.proto [hunk 4]
```diff
@@ -909,6 +909,8 @@ message StatusResponse {
   uint64 raftTerm = 6;
   // raftAppliedIndex is the current raft applied index of the responding member.
   uint64 raftAppliedIndex = 7;
+  // errors contains alarm/health information and status
```
Comment (xiang90) [inline etcdserver/etcdserverpb/rpc.proto]: add a dot `.` at the end of the sentence.

File: etcdserver/api/v3rpc/maintenance.go (hunk 3 above)
Comment (dvonthenen) [reply etcdserver/api/v3rpc/maintenance.go]: done

File: etcdserver/etcdserverpb/rpc.proto (hunk 4 above)
Comment (dvonthenen) [reply etcdserver/etcdserverpb/rpc.proto]: done

**Comment by xiang90** [comment]
defer to @gyuho

File: etcdserver/api/v3rpc/maintenance.go [hunk 5]
```diff
@@ -38,6 +39,9 @@ type BackendGetter interface {
 }
 
 type Alarmer interface {
+	// Alarms is implemented in Server interface located in etcdserver/server.go
+	// It returns the a list of alarms present in the AlarmStore
```
Comment (gyuho) [inline etcdserver/api/v3rpc/maintenance.go]: `s/the a/a/`?

**Comment by dvonthenen** [comment]
@gyuho fixed the typo

**Comment by xiang90** [comment]
/cc @liggitt 

we decided to keep /heath simple and not break the pervious format. the more complicated fields are moved to a gRPC service, where versioning/compatibility is easier to deal with.

**Pull request merged**